xarray/tests/test_backends.py

```diff
@@ -1971,24 +1971,24 @@ def test_manual_chunk(self) -> None:
 
         # Using chunks = None should return non-chunked arrays
         open_kwargs: dict[str, Any] = {"chunks": None}
-        with pytest.warns(UserWarning, match="The specified chunks separate"):
-            with self.roundtrip(original, open_kwargs=open_kwargs) as actual:
-                for k, v in actual.variables.items():
-                    # only index variables should be in memory
-                    assert v._in_memory == (k in actual.dims)
-                    # there should be no chunks
-                    assert v.chunks is None
+        with self.roundtrip(original, open_kwargs=open_kwargs) as actual:
+            for k, v in actual.variables.items():
+                # only index variables should be in memory
+                assert v._in_memory == (k in actual.dims)
+                # there should be no chunks
+                assert v.chunks is None
 
         # uniform arrays
         for i in range(2, 6):
             rechunked = original.chunk(chunks=i)
             open_kwargs = {"chunks": i}
-            with self.roundtrip(original, open_kwargs=open_kwargs) as actual:
-                for k, v in actual.variables.items():
-                    # only index variables should be in memory
-                    assert v._in_memory == (k in actual.dims)
-                    # chunk size should be the same as rechunked
-                    assert v.chunks == rechunked[k].chunks
+            with pytest.warns(UserWarning, match="The specified chunks separate"):
```

Comment:
The intent with `pytest.warns` IIUC is to check that a specific warning is raised. I think here we just want to silence any warnings about chunks. Use `warnings` instead?

Reply:
you're right.